packages/puppeteer-core/src/common/bidi/NetworkManager.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2023 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
+
+import {EventEmitter, Handler} from '../EventEmitter.js';
+import {NetworkManagerEmittedEvents} from '../NetworkManager.js';
+
+import {Connection} from './Connection.js';
+import {FrameManager} from './FrameManager.js';
+import {HTTPRequest} from './HTTPRequest.js';
+import {HTTPResponse} from './HTTPResponse.js';
+
+/**
+ * @internal
+ */
+export class NetworkManager extends EventEmitter {
+  #connection: Connection;
+  #frameManager: FrameManager;
```

Comment:
so why do we need frameManager here? can we avoid checking if the context is known?